src/test/static/js-behaviors.js

```diff
@@ -22,3 +22,35 @@ Really.Really.Deep.Behavior = [
 */
 CustomBehaviorList =
     [SimpleBehavior, CustomNamedBehavior, Really.Really.Deep.Behavior];
+
+function unhoistedGenerateBehaviorBefore() {
+  return {
+    properties: {
+      generatedBy: {
+        value: 'unhoistedGenerateBehavior'
+      }
+    }
+  };
+}
+
+/**
+ * Call to a function declared before
+ * @polymerBehavior
```

Comment:
It doesn't look like you're checking for `@polymerBehavior` annotations. Do you get the same test results without them?

I'm worried about false positives here. Mind adding a number of examples of function declarations and calls that should not be matched?